Emby.Server.Implementations/ApplicationHost.cs

```diff
@@ -1509,10 +1511,10 @@ public string GetLocalApiUrl(IPAddress ipAddress)
         public string GetLocalApiUrl(ReadOnlySpan<char> host)
         {
             var url = new StringBuilder(64);
-            url.Append(EnableHttps ? "https://" : "http://")
+            url.Append(ListenWithHttps ? "https://" : "http://")
                 .Append(host)
                 .Append(':')
-                .Append(EnableHttps ? HttpsPort : HttpPort);
+                .Append(ListenWithHttps ? HttpsPort : HttpPort);
```

Comment:
sorry, but this isn't completely right, see https://github.com/jellyfin/jellyfin/pull/2798

Reply:
I'm not 100% sure since I can't test this right now, but to me it looks like this is correct and actually makes #2798 unnecessary. Here `ListenWithHttps` is true if (and only if) the server is actually listening on an HTTPS port (ignoring reverse proxy, etc). So there is no need to force ffmpeg to use HTTP.

Reply:
I'm sorry, but your assumption here is incorrect. :)

Consider this case:
* Server is set to "handled by reverse proxy" and not given a certificate, so it itself doesn't listen on https port
* The user really has a reverse proxy sitting on the same host as JF listening to that https port and redirecting traffic to JF http port
* Firewall explicitly blocks http port from accessing outside
* This function is used to compose the URL to be sent to some _remote_ client (not always LiveTV ffmpeg), so for _most_ cases that client _must_ use https port even though JF itself doesn't listen to it.

It's only LiveTV that we force to http port, because there's no guarantee that reverse proxy that JF assumes is running is listening on `127.0.0.1:https`.

Reply:
> This function is used to compose the URL to be sent to some remote client

Okay in this context that makes sense, I didn't realize that. The function name `GetLocalApiUrl` here is very confusing then. Maybe it would be clearer to have two methods  and `GetLocalApiUrl` (using `ListenWithHttps `) and `GetRemoteApiUrl` (using `CanConnectWithHttps`). Or rename to `GetApiUrl` and add a bool parameter `isLocal`. How does that sound?

Reply:
I think `Local` meant "API for this server". And I have already added a parameter to this function in my PR...

So we should do a joint approach here. I actually think that `isLocal` name and renaming the function make more sense than my own variable (`forceHttp`) name, so let's do this. You can incorporate my #2798 in your PR, it's rather small I think.

Reply:
I like the idea of separate methods - as (I think?) the we could also have 3 variations on host - External/Proxied domain, Local Network/Host IP and Internal/localhost.

Reply:
Do you have any usecases for "Local Network"? I have a case for "Host IP" (but public) for Chromecast, which uses Google DNS regardless of what the network it is within uses, so we'd better resolve the hostname ourselves and feed the Chromecast the IP, but it should probably be the external IP because who knows if CC is inside the LAN or outside...

Reply:
To me, Local Network was how it works now in the default config.

Here are the questions I see the server needing to answer:
1. Where do I tell clients I am (use custom domain/proxy)
2. Where do I tell chromecasts I am (IP of an already resolved domain/proxy)
3. Where do I think I am (IP that the server is bound to )
4. Where should I call my own services (http://127.0.0.1) - this may be more "Where should I tell local FFMPEG to find my content" (for Live TV)

Reply:
I think what I'll do is create a method with all the parameters necessary to answer those 4 questions @PrplHaz4 mentioned (or other combinations for future use cases). Then also add additional methods, with more specific and accurate names for the most common use cases, that call the base method with specific parameters.

@JustAMan Sure, should be easy enough to integrate #2798 here, it's pretty small. I'll try to find some time later today to work on this

Reply:
@JustAMan I took another look at this and I'm not sure I follow this:

> This function is used to compose the URL to be sent to some remote client (not always LiveTV ffmpeg), so for most cases that client must use https port even though JF itself doesn't listen to it

Two reasons I think this can't be true:

1. The function doesn't take into accuont `PublicHttpsPort` or `PublicHttpPort` which would be necessary to report the URL to use to access via reverse proxy. In fact, these two properties only exist to set up port forwarding and aren't used anywhere else.
2. Nowhere in configuration does the server keep track of the public domain name or ip address that you would need to use to generate a URL access via reverse proxy

As far as I can tell the server doesn't have the ability to report to a client how it can be accessed remotely from behind a reverse proxy. Which makes sense, why would the server report that? The client already has to know how to contact the server in order to get any information from it.

Regarding the questions from @PrplHaz4:

1. Where do I tell clients I am (use custom domain/proxy)
    Like I mentioned above, I don't think the server can currently respond to this question and if we are to add that functionality it would be out of scope for this PR, where I am just trying to simplify the existing code to make it more readable.
2. Where do I tell chromecasts I am (IP of an already resolved domain/proxy)
    Do you know how this currently works? Not familiar with this so I didn't know where to look.
3. Where do I think I am (IP that the server is bound to)
    Already supported with the existing implementation
4. Where should I call my own services (http://127.0.0.1) - this may be more "Where should I tell local FFMPEG to find my content" (for Live TV)
    Already supported with the existing implementation

Reply:
I consider this fixed with latest changes.